base/special/trig.jl

```diff
@@ -199,9 +199,8 @@ function sincos(x::T) where T<:Union{Float32, Float64}
 end
 
 _sincos(x::AbstractFloat) = sincos(x)
-_sincos(x) = (sin(x), cos(x))
-
-sincos(x) = _sincos(float(x))
+_sincos(x::Real) = (sin(x), cos(x))
+sincos(x::Real) = _sincos(float(x))
```

Comment:
Does this now disallow complex inputs? Currently they work as expected:
```
julia> sincos(1+1im)
(1.2984575814159773 + 0.6349639147847361im, 0.8337300251311491 - 0.9888977057628651im)
```
I'd make the type restriction `Number` rather than `Real`.

Reply:
Excellent point! I forgot that `sin` and `cos` are defined for Complex.

Reply:
Done!